Enformer.pm

```diff
@@ -0,0 +1,156 @@
+
+ The plugin can then be run as default:
+ ./vep -i variations.vcf --assembly GRCh38 --plugin Enformer,file=/path/to/Enformer/data.vcf.gz
+
+ or run with option to only retrieve the SAD score 
+ ./vep -i variations.vcf --assembly GRCh38 --plugin Enformer,file=/path/to/Enformer/data.vcf.gz,SAD=1 
+ 
+ or run with option to only retrieve the SAR score 
+ ./vep -i variations.vcf --assembly GRCh38 --plugin Enformer,file=/path/to/Enformer/data.vcf.gz,SAR=1 
+
+
+ The tabix utility must be installed in your path to use this plugin.
+ Check https://github.com/samtools/htslib.git for instructions.
+
+
+=cut
+
+package Enformer;
+
+use strict;
+use warnings; 
+
+use Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin;
+use base qw(Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin);
+
+sub new {
+  my $class = shift;
+  
+  my $self = $class->SUPER::new(@_);
+
+  $self->expand_left(0);
+  $self->expand_right(0);
+  $self->get_user_params();
+  
+  my $params = $self->params_to_hash();
+  
+  my $file;
+  if (!keys %$params) {
+    $file = $self->params->[0];
+    $params->{file} = $file;
+  } else {
+    $file = $params->{file};
+    $self->{SAD} = $params->{SAD} if (defined ($params->{SAD}));
+    $self->{SAR} = $params->{SAR} if (defined ($params->{SAR}));
+  } 
+  
+  $self->{params} = $params;
+  $self->add_file($file);
+  
+  my $assembly = $self->{config}->{assembly};
+
+  return $self;
+
+}
+
+sub feature_types {
+  return ['Transcript','RegulatoryFeature','MotifFeature','Intergenic'];
+}
+
+sub get_header_info {
+  return { Enformer => "Prediction tool to accurately identify variant impact on gene expression" };
+}
+
+sub run{
+  my ($self, $tva) = @_;
+  
+  my $vf = $tva->variation_feature;
+  my $allele = $tva->variation_feature_seq;
+  return {} unless $allele =~ /^[ACGT]$/;
+  
+
+  my ($vf_start, $vf_end) = ($vf->{start}, $vf->{end});
+  ($vf_start, $vf_end) = ($vf_end, $vf_start) if ($vf_start > $vf_end);
+
+  my ($res) = grep{
+    $_->{alt} eq $allele &&
+    $_->{start} == $vf_start &&
+    $_->{start} == $vf_end
+    } @{$self->get_data($vf->{chr}, $vf_start, $vf_end)};
+    
+  
+  return $res ? $res->{result} : {} if keys($self->{params}) == 1;
+
+  return $res ? {Enformer => $res->{SAD}} : {} if defined($self->{SAD});
+
+  return $res ? {Enformer => $res->{SAR}} : {} if defined($self->{SAR});
+}
+
+
+sub parse_data {
+  my ($self, $line) = @_;
+  my ($chr_data, $start, $snp, $ref, $alt, $qual, $filter, $data) = split("\t", $line);
+  
+  my ($chr) = $chr_data =~ /(\d+)/; # this is because the chromosome is chr1 etc, to retrieve just the 1
+  
+
+  my @data_splitted = split(";", $data); #splitting data in the vcf file based on the format using ;
+  my $SAD = $data_splitted[0]; # SAD comes first 
+  my $SAR = $data_splitted[1]; #SAR is second 
+
+  return {
+    chr => $chr,
+    start => $start,
+    ref => $ref,
+    alt => $alt,
+    SAD => $SAD,
+    SAR => $SAR,
+    result => {
+      Enformer => $data
+    }
```

Comment:
The plugin should directly return the results. Some plugins also put their name as a prefix, such as done in [EVE plugin](https://github.com/Ensembl/VEP_plugins/blob/e0499feee1b5502499f3aa3c4539b6fda4566962/EVE.pm#L190-L199) (easier for users when using multiple plugins).
```suggestion
    result => {
      Enformer_SAD => $SAD,
      Enformer_SAR => $SAR
    }
```